os/arch/arm/src/imxrt/imxrt_gpio.c

```diff
@@ -729,18 +729,52 @@ static inline int imxrt_gpio_configperiph(gpio_pinset_t pinset)
 /****************************************************************************
  * Public Functions
  ****************************************************************************/
-uint8_t imxrt_gpio_irqvector(uint32_t pinset)
+uint16_t imxrt_gpio_irqvector(uint32_t pinset)
 {
 	int port = (pinset & GPIO_PORT_MASK) >> GPIO_PORT_SHIFT;
 	int pin = (pinset & GPIO_PIN_MASK) >> GPIO_PIN_SHIFT;
+	uint16_t ret = 0;
 
-	if (port == GPIO_PORT1 >> GPIO_PORT_SHIFT) {
-		return _IMXRT_GPIO1_0_15_BASE + pin;
-	} else if (port == GPIO_PORT2 >> GPIO_PORT_SHIFT) {
-		return _IMXRT_GPIO1_16_31_BASE + pin;
+	switch (port) {
+	case GPIO1INDEX:
+		if (pin > 15) {
+			ret = _IMXRT_GPIO1_16_31_BASE + (pin - 16);
```

Comment:
We don't have to check CONFIG_IMXRT_GPIO1_0_15_IRQ & CONFIG_IMXRT_GPIO1_16_31_IRQ?
Base value is changed based on these config.

Reply:
The base value in GPIO irq will be changed in previous step, imxrt102x_irq.h/imxrt105x_irg.h related to CONFIG_IMXRT_GPIO1_0_15_IRQ, ... CONFIG_IMXRT_GPIO5_16_31_IRQ

Reply:
ok, got it!.